modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java

```diff
@@ -752,6 +774,173 @@ public void testIdleVerifyOnInactiveClusterWithPersistence() throws Exception {
         assertContains(log, testOut.toString(), "The check procedure has finished, no conflicts have been found.");
     }
 
+    /**
+     *
+     */
+    @Test
+    public void testIdleVerifyCancelCommandOnCheckpoint() throws Exception {
+        final int gridsCnt = 4;
+
+        IgniteEx srv = startGrids(gridsCnt);
+
+        srv.cluster().state(ACTIVE);
+
+        CountDownLatch beforeCancelLatch = new CountDownLatch(1);
+        CountDownLatch afterCancelLatch = new CountDownLatch(1);
+
+        GridCacheDatabaseSharedManager dbMgr = (GridCacheDatabaseSharedManager)grid(1).context().cache().context().database();
+
+        dbMgr.addCheckpointListener(new CheckpointListener() {
+            @Override public void beforeCheckpointBegin(Context ctx) {
+                if (ctx.progress().reason().equals("VerifyBackupPartitions"))
+                    beforeCancelLatch.countDown();
+            }
+
+            @Override public void afterCheckpointEnd(Context ctx) throws IgniteCheckedException {
+                if (ctx.progress().reason().equals("VerifyBackupPartitions")) {
+                    try {
+                        afterCancelLatch.await(30, TimeUnit.SECONDS);
+                    }
+                    catch (InterruptedException e) {
+                        throw new IgniteInterruptedCheckedException(e);
+                    }
+                }
+            }
+
+            @Override public void onMarkCheckpointBegin(Context ctx) {
+                // No-op.
+            }
+
+            @Override public void onCheckpointBegin(Context ctx) {
+                // No-op.
+            }
+        });
+
+        List<LogListener> listeners = registerIdleVerifyCancelListeners();
```

Comment:
Looks like registering and checking listeners can be moved to `cancelIdleVerifyAndCheck`